core/market/decentralized/src/rest_api/requestor.rs

```diff
@@ -104,12 +104,23 @@ async fn counter_proposal(
 async fn get_proposal(
     market: Data<Arc<MarketService>>,
     path: Path<PathSubscriptionProposal>,
-    id: Identity,
+    _id: Identity,
 ) -> impl Responder {
-    let PathSubscriptionProposal { proposal_id, .. } = path.into_inner();
+    // TODO: Authorization
+    let PathSubscriptionProposal {
+        subscription_id,
+        proposal_id,
+    } = path.into_inner();
+
+    // check if subscription exists
+    market.matcher.store.get_demand(&subscription_id).await?;
```

Comment:
You can't view historical `Proposals` because of this.
I wouldn't check this, because you can't do anything wrong with `Proposals` for expired Offers/Demands.

Reply:
I was aware of it, but after reconsideration, I'm removing this check.